### PR TITLE
Added grades publishing feature.

### PR DESCRIPTION
**Pull request by iivic**
#NDPD-832
screen shot 2018-01-19 at 09 30 05

### CODE REVIEW

File: edx_sga/sga.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import os
 import pkg_resources
 import pytz
+from distutils.util import strtobool
```
Comment (marjev) [inline edx_sga/sga.py]: nitpick: order of imports

File: edx_sga/sga.py [hunk 2]
```diff
@@ -356,6 +366,11 @@ def save_sga(self, data, suffix=''):
             raise JsonHandlerError(400, 'Points must be a positive integer')
         self.points = points
 
+    @XBlock.handler
+    def update_grades_published(self, request, suffix=''):
+        self.grades_published = bool(strtobool(request.params.get('grades_published')))
```
Comment (marjev) [inline edx_sga/sga.py]: `json.loads(request.params.get('grades_published'))`

File: edx_sga/static/js/src/edx_sga.js [hunk 3]
```diff
@@ -1,27 +1,40 @@
 /* Javascript for StaffGradedAssignmentXBlock. */
-function StaffGradedAssignmentXBlock(runtime, element) {
+function StaffGradedAssignmentXBlock(runtime, element, backendData) {
```
Comment (marjev) [inline edx_sga/static/js/src/edx_sga.js]: Call this `options` or something more generic, other than backendData. You named it to describe where it comes from, but not what it's expected to have or do. `runtime` and `element` are generic, but descriptive enough if you're familiar with the concept of an XBlock.

File: edx_sga/static/js/src/edx_sga.js [hunk 4]
```diff
@@ -1,27 +1,40 @@
...
+        var removeGradeUrl = runtime.handlerUrl(element, 'remove_grade');
         var staffAnnotatedUrl = runtime.handlerUrl(element, 'staff_download_annotated');
+        var staffDownloadUrl = runtime.handlerUrl(element, 'staff_download');
         var staffUploadUrl = runtime.handlerUrl(element, 'staff_upload_annotated');
-        var enterGradeUrl = runtime.handlerUrl(element, 'enter_grade');
-        var removeGradeUrl = runtime.handlerUrl(element, 'remove_grade');
         var template = _.template($(element).find("#sga-tmpl").text());
-        var gradingTemplate;
+        var uploadUrl = runtime.handlerUrl(element, 'upload_assignment');
+        var updateGradesPublishedUrl = runtime.handlerUrl(element, 'update_grades_published')
```
Comment (marjev) [inline edx_sga/static/js/src/edx_sga.js]: nitpick: alphabetize

File: edx_sga/templates/staff_graded_assignment/show.html [hunk 5]
```diff
@@ -117,7 +117,13 @@
   <section aria-hidden="true" class="modal staff-modal" id="{{ id }}-grade" style="height: 75%">
     <div class="inner-wrapper" style="color: black; overflow: auto;">
       <header><h2>{% trans "Staff Graded Assignment" %}</h2></header>
-      <br/>
+      <br>
+
+      <div class="grades_published">
+        <label for="grades-published">Grades published:</label>
+        <input id="grades-published" name="grades_published" type="checkbox" {% if grades_published %} checked {% endif %}>
+      </div>
```
Comment (marjev) [inline edx_sga/templates/staff_graded_assignment/show.html]: https://www.google.hr/search?client=firefox-b-ab&dcr=0&biw=1280&bih=615&tbm=isch&sa=1&ei=V2RgWoeVOMmlsgHFiYa4Ag&q=checkbox+labels&oq=checkbox+labels&gs_l=psy-ab.3..0i24k1.12613.14293.0.14956.9.6.1.2.2.0.175.783.1j5.6.0....0...1c.1.64.psy-ab..0.9.803...0j0i5i30k1j0i8i30k1j0i13k1.0.zqqrH_G6A_c#imgrc=tTFAUIslSjY1cM:


Notice the difference.

**Review by marjev — CHANGES_REQUESTED** [review]
Also fix the conflicts and bump the version of the package.

**Comment by iivic** [comment]
@marjev Please take another look.

**Review by marjev — APPROVED** [review]

**Pull request merged**